File: VB/4PosBackOffice.NET/frmOrderItemQuick.vb
Option Strict Off
Option Explicit On
Friend Class frmOrderItemQuick
	Inherits System.Windows.Forms.Form
	Dim adoPrimaryRS As ADODB.Recordset
	
	Private Sub loadLanguage()
		
		'frmOrderItemQuick = No Code      [GRV Quick Edit]
		'rsLang.filter = "LanguageLayoutLnk_LanguageID=" & 0000
		'If rsLang.RecordCount Then frmOrderItemQuick.Caption = rsLang("LanguageLayoutLnk_Description"): frmOrderItemQuick.RightToLeft = rsLang("LanguageLayoutLnk_RightTL")
		
		'lblName = No Code/Dynamic/NA?
		
		'chkBreakPack = No Code         [Break Pack]
		'rsLang.filter = "LanguageLayoutLnk_LanguageID=" & 0000
		'If rsLang.RecordCount Then chkBreakPack.Caption = rsLang("LanguageLayoutLnk_Description"): chkBreakPack.RightToLeft = rsLang("LanguageLayoutLnk_RightTL")
		
		'_lbl_0 = No Code               [Quantity]
		'rsLang.filter = "LanguageLayoutLnk_LanguageID=" & 0000
		'If rsLang.RecordCount Then _lbl_0.Caption = rsLang("LanguageLayoutLnk_Description"): _lbl_0.RightToLeft = rsLang("LanguageLayoutLnk_RightTL")
		
		'    rsLang.filter = "LanguageLayoutLnk_LanguageID=" & 1151 'Price|Checked
		'    If rsLang.RecordCount Then _lbl_2.Caption = rsLang("LanguageLayoutLnk_Description"): _lbl_2.RightToLeft = rsLang("LanguageLayoutLnk_RightTL")
		'
		'    rsLang.filter = "LanguageLayoutLnk_LanguageID=" & 1662 'Discount|Checked
		'    If rsLang.RecordCount Then Label1.Caption = rsLang("LanguageLayoutLnk_Description"): Label1.RightToLeft = rsLang("LanguageLayoutLnk_RightTL")
		'
		'_lbl_1 = No Code               [Surcharges]
		'rsLang.filter = "LanguageLayoutLnk_LanguageID=" & 0000
		'If rsLang.RecordCount Then _lbl_1.Caption = rsLang("LanguageLayoutLnk_Description"): _lbl_1.RightToLeft = rsLang("LanguageLayoutLnk_RightTL")
		
		'    rsLang.filter = "LanguageLayoutLnk_LanguageID=" & 1217 'Proceed|Checked
		'    If rsLang.RecordCount Then cmdProceed.Caption = rsLang("LanguageLayoutLnk_Description"): cmdProceed.RightToLeft = rsLang("LanguageLayoutLnk_RightTL")
		'
		'    rsHelp.filter = "Help_Section=0 AND Help_Form='" & Me.Name & "'"
		'    If rsHelp.RecordCount Then Me.ToolTip1 = rsHelp("Help_ContextID")
		
	End Sub
	
    Private Sub chkBreakPack_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs)
        'If chkBreakPack.value Then
        ' Me.txtMin.Tag = adoPrimaryRS.Fields("StockItem_UnitCost").Value
        ' Else
        ' Me.txtMin.Tag = adoPrimaryRS.Fields("StockItem_ListCost").Value
        ' End If

        Select Case frmGRVitem.activePrice
            Case 0
                Me.txtMin.Text = FormatNumber(Me.txtMin.Tag, 2)
            Case 1
                Me.txtMin.Text = FormatNumber(CDec(txtMin.Tag) * (1 + CDbl(frmGRVitem.gridItem.get_TextMatrix(frmGRVitem.gridItem.row, frmGRVitem.colVAT)) / 100), 2)

            Case 2
                Me.txtMin.Text = FormatNumber(CDec(Me.txtMin.Tag) + CDec(frmGRVitem.gridItem.get_TextMatrix(frmGRVitem.gridItem.row, frmGRVitem.colDepositExclusive)), 2)
            Case 3
                Me.txtMin.Text = FormatNumber((CDec(Me.txtMin.Tag) + CDec(frmGRVitem.gridItem.get_TextMatrix(frmGRVitem.gridItem.row, frmGRVitem.colDepositExclusive))) * (1 + CDbl(frmGRVitem.gridItem.get_TextMatrix(frmGRVitem.gridItem.row, frmGRVitem.colVAT)) / 100), 2)
        End Select
    End Sub
	
	Private Sub chkBreakPack_KeyPress(ByRef KeyAscii As Short)
		If KeyAscii = 27 Then
			Me.Close()
		End If
		
	End Sub
	
	Private Sub cmdProceed_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdProceed.Click
        Dim sql As String
		Dim dirty As Boolean
		dirty = False
		Dim lPrice, lDiscount As Decimal
		
		If CDbl(txtCost.Tag) <> CDec(txtCost.Text) Then dirty = True
		If CDbl(txtMin.Tag) <> CShort(txtMin.Text) Then dirty = True
		If CDbl(txtMax.Tag) <> CShort(txtMax.Text) Then dirty = True
		
		If dirty Then
			sql = "UPDATE MinMaxStockItemLnk SET MinMaxStockItemLnk_Minimum = " & CShort(txtMin.Text) & " Where (MinMaxStockItemLnk_MinMaxID = 1) And (MinMaxStockItemLnk_StockItemID = " & CShort(Me.lblName.Tag) & ")"
			cnnDB.Execute(sql)
			
			cnnDB.Execute("UPDATE StockItem SET StockItem.StockItem_MinimumStock = " & CShort(txtMin.Text) & ", StockItem_OrderRounding = " & CShort(txtMax.Text) & " WHERE StockItemID = " & CShort(Me.lblName.Tag) & ";")
		End If
		Me.Close()
	End Sub
	
	Private Sub frmOrderItemQuick_Load(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles MyBase.Load
		If frmOrderWizard.Tag <> "" Then
			loadLanguage()
			
			Me.lblName.Tag = Split(frmOrderWizard.Tag, "|")(0)
			Me.lblName.Text = Split(frmOrderWizard.Tag, "|")(1)
			
			Me.txtCost.Text = Split(frmOrderWizard.Tag, "|")(2)
			Me.txtMin.Text = Split(frmOrderWizard.Tag, "|")(4)
			Me.txtMax.Text = Split(frmOrderWizard.Tag, "|")(3)
			
			txtCost.Tag = CDec(txtCost.Text)
			txtMin.Tag = CDec(txtMin.Text)
			txtMax.Tag = CDec(txtMax.Text)
			
		Else
			Me.Close()
		End If
	End Sub
	
	Private Sub txtMax_Enter(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles txtMax.Enter
        MyGotFocusNumeric(txtMax)
	End Sub
	
	Private Sub txtMax_KeyPress(ByVal eventSender As System.Object, ByVal eventArgs As System.Windows.Forms.KeyPressEventArgs) Handles txtMax.KeyPress
		Dim KeyAscii As Short = Asc(eventArgs.KeyChar)
		If KeyAscii = 27 Then
			Me.Close()
		Else
			MyKeyPress(KeyAscii)
		End If
		eventArgs.KeyChar = Chr(KeyAscii)
		If KeyAscii = 0 Then
			eventArgs.Handled = True
		End If
	End Sub
	
	Private Sub txtMax_Leave(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles txtMax.Leave
        MyLostFocus(txtMax, 0)
	End Sub
	Private Sub txtMin_Enter(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles txtMin.Enter
        MyGotFocusNumeric(txtMin)
	End Sub
	
	Private Sub txtMin_KeyPress(ByVal eventSender As System.Object, ByVal eventArgs As System.Windows.Forms.KeyPressEventArgs) Handles txtMin.KeyPress
		Dim KeyAscii As Short = Asc(eventArgs.KeyChar)
		If KeyAscii = 27 Then
			Me.Close()
		Else
            MyKeyPress(KeyAscii)
		End If
		eventArgs.KeyChar = Chr(KeyAscii)
		If KeyAscii = 0 Then
			eventArgs.Handled = True
		End If
	End Sub
	
	Private Sub txtMin_Leave(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles txtMin.Leave
        MyLostFocus(txtMin, 0)
	End Sub
	
	Private Sub txtCost_Enter(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles txtCost.Enter
        MyGotFocusNumeric(txtCost)
	End Sub
	
	Private Sub txtCost_KeyPress(ByVal eventSender As System.Object, ByVal eventArgs As System.Windows.Forms.KeyPressEventArgs) Handles txtCost.KeyPress
		Dim KeyAscii As Short = Asc(eventArgs.KeyChar)
		If KeyAscii = 27 Then
			Me.Close()
		Else
            MyKeyPress(KeyAscii)
		End If
		eventArgs.KeyChar = Chr(KeyAscii)
		If KeyAscii = 0 Then
			eventArgs.Handled = True
		End If
	End Sub
	
	Private Sub txtCost_Leave(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles txtCost.Leave
        MyLostFocus(txtCost, 2)
	End Sub
End Class

File: C-Sharp/frmOrderItemQuick.cs
using CrystalDecisions.CrystalReports.Engine;
using CrystalDecisions.Windows.Forms;
using DpSdkEngLib;
using DPSDKOPSLib;
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Linq;
using System.Xml.Linq;
 // ERROR: Not supported in C#: OptionDeclaration
namespace _4PosBackOffice.NET
{
	internal partial class frmOrderItemQuick : System.Windows.Forms.Form
	{
		ADODB.Recordset adoPrimaryRS;

		private void loadLanguage()
		{

			//frmOrderItemQuick = No Code      [GRV Quick Edit]
			//rsLang.filter = "LanguageLayoutLnk_LanguageID=" & 0000
			//If rsLang.RecordCount Then frmOrderItemQuick.Caption = rsLang("LanguageLayoutLnk_Description"): frmOrderItemQuick.RightToLeft = rsLang("LanguageLayoutLnk_RightTL")

			//lblName = No Code/Dynamic/NA?

			//chkBreakPack = No Code         [Break Pack]
			//rsLang.filter = "LanguageLayoutLnk_LanguageID=" & 0000
			//If rsLang.RecordCount Then chkBreakPack.Caption = rsLang("LanguageLayoutLnk_Description"): chkBreakPack.RightToLeft = rsLang("LanguageLayoutLnk_RightTL")

			//_lbl_0 = No Code               [Quantity]
			//rsLang.filter = "LanguageLayoutLnk_LanguageID=" & 0000
			//If rsLang.RecordCount Then _lbl_0.Caption = rsLang("LanguageLayoutLnk_Description"): _lbl_0.RightToLeft = rsLang("LanguageLayoutLnk_RightTL")

			//    rsLang.filter = "LanguageLayoutLnk_LanguageID=" & 1151 'Price|Checked
			//    If rsLang.RecordCount Then _lbl_2.Caption = rsLang("LanguageLayoutLnk_Description"): _lbl_2.RightToLeft = rsLang("LanguageLayoutLnk_RightTL")
			//
			//    rsLang.filter = "LanguageLayoutLnk_LanguageID=" & 1662 'Discount|Checked
			//    If rsLang.RecordCount Then Label1.Caption = rsLang("LanguageLayoutLnk_Description"): Label1.RightToLeft = rsLang("LanguageLayoutLnk_RightTL")
			//
			//_lbl_1 = No Code               [Surcharges]
			//rsLang.filter = "LanguageLayoutLnk_LanguageID=" & 0000
			//If rsLang.RecordCount Then _lbl_1.Caption = rsLang("LanguageLayoutLnk_Description"): _lbl_1.RightToLeft = rsLang("LanguageLayoutLnk_RightTL")

			//    rsLang.filter = "LanguageLayoutLnk_LanguageID=" & 1217 'Proceed|Checked
			//    If rsLang.RecordCount Then cmdProceed.Caption = rsLang("LanguageLayoutLnk_Description"): cmdProceed.RightToLeft = rsLang("LanguageLayoutLnk_RightTL")
			//
			//    rsHelp.filter = "Help_Section=0 AND Help_Form='" & Me.Name & "'"
			//    If rsHelp.RecordCount Then Me.ToolTip1 = rsHelp("Help_ContextID")

		}

		private void chkBreakPack_Click(System.Object eventSender, System.EventArgs eventArgs)
		{
			//If chkBreakPack.value Then
			// Me.txtMin.Tag = adoPrimaryRS.Fields("StockItem_UnitCost").Value
			// Else
			// Me.txtMin.Tag = adoPrimaryRS.Fields("StockItem_ListCost").Value
			// End If

			switch (My.MyProject.Forms.frmGRVitem.activePrice) {
				case 0:
					this.txtMin.Text = Strings.FormatNumber(this.txtMin.Tag, 2);
					break;
				case 1:
					this.txtMin.Text = Strings.FormatNumber(Convert.ToDecimal(txtMin.Tag) * (1 + Convert.ToDouble(My.MyProject.Forms.frmGRVitem.gridItem.get_TextMatrix(ref My.MyProject.Forms.frmGRVitem.gridItem.row, ref My.MyProject.Forms.frmGRVitem.colVAT)) / 100), 2);

					break;
				case 2:
					this.txtMin.Text = Strings.FormatNumber(Convert.ToDecimal(this.txtMin.Tag) + Convert.ToDecimal(My.MyProject.Forms.frmGRVitem.gridItem.get_TextMatrix(ref My.MyProject.Forms.frmGRVitem.gridItem.row, ref My.MyProject.Forms.frmGRVitem.colDepositExclusive)), 2);
					break;
				case 3:
					this.txtMin.Text = Strings.FormatNumber((Convert.ToDecimal(this.txtMin.Tag) + Convert.ToDecimal(My.MyProject.Forms.frmGRVitem.gridItem.get_TextMatrix(ref My.MyProject.Forms.frmGRVitem.gridItem.row, ref My.MyProject.Forms.frmGRVitem.colDepositExclusive))) * (1 + Convert.ToDouble(My.MyProject.Forms.frmGRVitem.gridItem.get_TextMatrix(ref My.MyProject.Forms.frmGRVitem.gridItem.row, ref My.MyProject.Forms.frmGRVitem.colVAT)) / 100), 2);
					break;
			}
		}

		private void chkBreakPack_KeyPress(ref short KeyAscii)
		{
			if (KeyAscii == 27) {
				this.Close();
			}

		}

		private void cmdProceed_Click(System.Object eventSender, System.EventArgs eventArgs)
		{
			string sql = null;
			bool dirty = false;
			dirty = false;
			decimal lPrice = default(decimal);
			decimal lDiscount = default(decimal);

			if (Convert.ToDouble(txtCost.Tag) != Convert.ToDecimal(txtCost.Text))
				dirty = true;
			if (Convert.ToDouble(txtMin.Tag) != Convert.ToInt16(txtMin.Text))
				dirty = true;
			if (Convert.ToDouble(txtMax.Tag) != Convert.ToInt16(txtMax.Text))
				dirty = true;

			if (dirty) {
				sql = "UPDATE MinMaxStockItemLnk SET MinMaxStockItemLnk_Minimum = " + Convert.ToInt16(txtMin.Text) + " Where (MinMaxStockItemLnk_MinMaxID = 1) And (MinMaxStockItemLnk_StockItemID = " + Convert.ToInt16(this.lblName.Tag) + ")";
				modRecordSet.cnnDB.Execute(sql);

				modRecordSet.cnnDB.Execute("UPDATE StockItem SET StockItem.StockItem_MinimumStock = " + Convert.ToInt16(txtMin.Text) + ", StockItem_OrderRounding = " + Convert.ToInt16(txtMax.Text) + " WHERE StockItemID = " + Convert.ToInt16(this.lblName.Tag) + ";");
			}
			this.Close();
		}

		private void frmOrderItemQuick_Load(System.Object eventSender, System.EventArgs eventArgs)
		{
			if (!string.IsNullOrEmpty(My.MyProject.Forms.frmOrderWizard.Tag)) {
				loadLanguage();

				this.lblName.Tag = Strings.Split(My.MyProject.Forms.frmOrderWizard.Tag, "|")[0];
				this.lblName.Text = Strings.Split(My.MyProject.Forms.frmOrderWizard.Tag, "|")[1];

				this.txtCost.Text = Strings.Split(My.MyProject.Forms.frmOrderWizard.Tag, "|")[2];
				this.txtMin.Text = Strings.Split(My.MyProject.Forms.frmOrderWizard.Tag, "|")[4];
				this.txtMax.Text = Strings.Split(My.MyProject.Forms.frmOrderWizard.Tag, "|")[3];

				txtCost.Tag = Convert.ToDecimal(txtCost.Text);
				txtMin.Tag = Convert.ToDecimal(txtMin.Text);
				txtMax.Tag = Convert.ToDecimal(txtMax.Text);

			} else {
				this.Close();
			}
		}

		private void txtMax_Enter(System.Object eventSender, System.EventArgs eventArgs)
		{
			modUtilities.MyGotFocusNumeric(ref txtMax);
		}

		private void txtMax_KeyPress(System.Object eventSender, System.Windows.Forms.KeyPressEventArgs eventArgs)
		{
			short KeyAscii = Strings.Asc(eventArgs.KeyChar);
			if (KeyAscii == 27) {
				this.Close();
			} else {
				modUtilities.MyKeyPress(ref KeyAscii);
			}
			eventArgs.KeyChar = Strings.Chr(KeyAscii);
			if (KeyAscii == 0) {
				eventArgs.Handled = true;
			}
		}

		private void txtMax_Leave(System.Object eventSender, System.EventArgs eventArgs)
		{
			modUtilities.MyLostFocus(ref txtMax, ref 0);
		}
		private void txtMin_Enter(System.Object eventSender, System.EventArgs eventArgs)
		{
			modUtilities.MyGotFocusNumeric(ref txtMin);
		}

		private void txtMin_KeyPress(System.Object eventSender, System.Windows.Forms.KeyPressEventArgs eventArgs)
		{
			short KeyAscii = Strings.Asc(eventArgs.KeyChar);
			if (KeyAscii == 27) {
				this.Close();
			} else {
				modUtilities.MyKeyPress(ref KeyAscii);
			}
			eventArgs.KeyChar = Strings.Chr(KeyAscii);
			if (KeyAscii == 0) {
				eventArgs.Handled = true;
			}
		}

		private void txtMin_Leave(System.Object eventSender, System.EventArgs eventArgs)
		{
			modUtilities.MyLostFocus(ref txtMin, ref 0);
		}

		private void txtCost_Enter(System.Object eventSender, System.EventArgs eventArgs)
		{
			modUtilities.MyGotFocusNumeric(ref txtCost);
		}

		private void txtCost_KeyPress(System.Object eventSender, System.Windows.Forms.KeyPressEventArgs eventArgs)
		{
			short KeyAscii = Strings.Asc(eventArgs.KeyChar);
			if (KeyAscii == 27) {
				this.Close();
			} else {
				modUtilities.MyKeyPress(ref KeyAscii);
			}
			eventArgs.KeyChar = Strings.Chr(KeyAscii);
			if (KeyAscii == 0) {
				eventArgs.Handled = true;
			}
		}

		private void txtCost_Leave(System.Object eventSender, System.EventArgs eventArgs)
		{
			modUtilities.MyLostFocus(ref txtCost, ref 2);
		}
	}
}
